### PR TITLE
OMWorld: Removal of flags, general cleanups

### PR DESCRIPTION
**Pull request by xmas92**
Cleanups in preparation for opening up a PR to mainline for [JDK-8315884](https://bugs.openjdk.org/browse/JDK-8315884).

@coleenp I incorporated #182 into this.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/lilliput.git pull/183/head:pull/183` \
`$ git checkout pull/183`

Update a local copy of the PR: \
`$ git checkout pull/183` \
`$ git pull https://git.openjdk.org/lilliput.git pull/183/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 183`

View PR using the GUI difftool: \
`$ git pr show -t 183`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/lilliput/pull/183.diff">https://git.openjdk.org/lilliput/pull/183.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/lilliput/pull/183#issuecomment-2180503447)

### CODE REVIEW

**Comment by xmas92** [comment]
I will do a rebase on JDK-24 next week after getting these in. 

Already did one earlier this week. But I felt like it was better to get this OMWorld cleanups in. (I believe @coleenp is about done with the initial pre-review of it.) But one not on the next rebase, as parallel full GC has been changed to work more like G1 it will also require adaption with to use the Alt Full GC forwarding. E.g. from my earlier rebase ae4715b7d7ce6297d939b4eba5525581765a7b95

File: src/hotspot/share/utilities/vmError.cpp [hunk 1]
```diff
@@ -590,9 +590,6 @@ static void report_vm_version(outputStream* st, char* buf, int buflen) {
 #endif
                  UseCompressedOops ? ", compressed oops" : "",
                  UseCompressedClassPointers ? ", compressed class ptrs" : "",
-                 LockingMode == LM_MONITOR ? ", lm_monitors" :
-                 LockingMode == LM_LEGACY ? ", lm_legacy" :
-                 LockingMode == LM_LIGHTWEIGHT ? ", lm_lightweight" : "",
```
Comment (coleenp) [inline src/hotspot/share/utilities/vmError.cpp:595]: Without this, I guess the only way to know which locking mode is on is via. the command line in the hs_err file.  It didn't really belong here, but maybe we want to find somewhere else for it.  Not with this change.  It's good to clean this up for this change.

File: src/hotspot/share/runtime/objectMonitor.cpp [hunk 2]
```diff
@@ -296,10 +296,10 @@ void ObjectMonitor::ClearSuccOnSuspend::operator()(JavaThread* current) {
   }
 }
 
-#define assert_mark_word_concistency()                                                 \
+#define assert_mark_word_concistency()                                         \
```
Comment (coleenp) [inline src/hotspot/share/runtime/objectMonitor.cpp:299]: While you're here. The spelling is consistency.

File: src/hotspot/share/runtime/globals.hpp [hunk 3]
```diff
@@ -1977,23 +1977,13 @@ const int ObjectAlignmentInBytes = 8;
...
-  product(bool, OMC2UnrollCacheLookupLoopTail, true, "")                    \
-                                                                            \
-  product(int, OMC2UnrollCacheEntries, 0, "")                               \
-          range(0, OMCache::CAPACITY)                                       \
-                                                                            \
-  product(int, OMCacheSize, 8, "")                                          \
-          range(0, OMCache::CAPACITY)                                       \
-                                                                            \
-  product(int, OMSpins, 13,                                                 \
+  product(int, LightweightFastLockingSpins, 13, DIAGNOSTIC,                 \
```
Comment (coleenp) [inline src/hotspot/share/runtime/globals.hpp:1980]: Good.  I like the new name.

File: src/hotspot/cpu/x86/sharedRuntime_x86.cpp [hunk 4]
```diff
@@ -62,9 +62,11 @@ void SharedRuntime::inline_check_hashcode_from_object_header(MacroAssembler* mas
...
 
   if (LockingMode == LM_LIGHTWEIGHT) {
-    // check if monitor
-    __ testptr(result, markWord::monitor_value);
-    __ jcc(Assembler::notZero, slowCase);
+    if (!UseObjectMonitorTable) {
+      // check if monitor
+      __ testptr(result, markWord::monitor_value);
+      __ jcc(Assembler::notZero, slowCase);
+    }
```
Comment (coleenp) [inline src/hotspot/cpu/x86/sharedRuntime_x86.cpp:69]: Did I miss this one?

**Review by coleenp — APPROVED** [review]
The cleanups look really good.  Can you fix the spelling of consistency?

File: src/hotspot/share/utilities/vmError.cpp (hunk 1 above)
Comment (xmas92) [reply src/hotspot/share/utilities/vmError.cpp:595]: We print flags at the end which are non-default. But we do not know if someone changed them in a build. 
Should probably fix this somehow. 

Also printing all flag values, not just non-default ones would help.

File: src/hotspot/cpu/x86/sharedRuntime_x86.cpp (hunk 4 above)
Comment (xmas92) [reply src/hotspot/cpu/x86/sharedRuntime_x86.cpp:69]: We did the C2 one. This is the native wrapper one. We both overlooked it.

Comment (coleenp) [reply src/hotspot/cpu/x86/sharedRuntime_x86.cpp:69]: I see.  We missed an optimization here.

**Review by coleenp — APPROVED** [review]

**Review by coleenp — APPROVED** [review]

**Comment by xmas92** [comment]
Thanks for the review.
/integrate

**Pull request closed without merging**